pkg/test/framework/components/ingress/kube.go

```diff
@@ -120,28 +157,92 @@ func (c *kubeComponent) Address() string {
 	return c.address
 }
 
-func (c *kubeComponent) Call(path string) (CallResponse, error) {
+// createClient creates a client which sends HTTP requests or HTTPS requests, depending on
+// ingress type. If host is not empty, the client will resolve domain name and verify server
+// cert using the host name.
+func (c *kubeComponent) createClient(host string) (*http.Client, error) {
 	client := &http.Client{
 		Timeout: 1 * time.Minute,
 	}
+	if c.gatewayType != PlainText {
+		scopes.Framework.Debug("Prepare root cert for client")
+		roots := x509.NewCertPool()
+		ok := roots.AppendCertsFromPEM([]byte(c.caCert))
+		if !ok {
+			return nil, errors.New("failed to parse root certificate")
+		}
+		tlsConfig := &tls.Config{
+			RootCAs:    roots,
+			ServerName: host,
+		}
+		if c.gatewayType == Mtls {
+			cer, err := tls.X509KeyPair([]byte(c.tlsCert), []byte(c.tlsKey))
+			if err != nil {
+				return nil, errors.New("failed to parse private key and server cert")
+			}
+			tlsConfig.Certificates = []tls.Certificate{cer}
+		}
+		tr := &http.Transport{
+			TLSClientConfig: tlsConfig,
+			DialTLS: func(netw, addr string) (net.Conn, error) {
+				if addr == host+":443" {
+					addr = c.address + ":443"
+				}
```

Comment:
What does this logic mean?

Reply:
It means when test client sends https://bookinfo.example.com, the function helps to resolve to IP address. Otherwise, TLS dial would fail. This is similar to curl --resolve

Reply:
I see. Does it make sense to rename c.address to c.ipAddress?